File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/interpreter/booleanoperation/AndOperation.java
package com.bigboxcode.designpattern.interpreter.booleanoperation;

public class AndOperation implements Operation {

    private final Operation op1;
    private final Operation op2;

    public AndOperation(Operation op1, Operation op2) {
        this.op1 = op1;
        this.op2 = op2;
    }

    @Override
    public boolean execute(String opContext) {
        return op1.execute(opContext) && op2.execute(opContext);
    }
}

File: design-patterns/php/interpreter/logicalop/AndOperation.php
<?php
// AndOperation.php

namespace BigBoxCode\DesignPattern\Interpreter\LogicalOp;

class AndOperation implements Operation {
    public function __construct(private Operation $op1, private Operation $op2) {
    }

    public function execute(string $opContext): bool {
        return $this->op1->execute($opContext) && $this->op2->execute($opContext);
    }
}